imod/mf6/simulation.py

```diff
@@ -648,40 +668,56 @@ def _merge_budgets(
 
         return merge_partitions(cbc_per_partition)
 
-    def _concat_concentrations(
-        self, modelnames: list[str], species_ls: list[str], output: str, **settings
-    ) -> GridDataArray:
-        concentrations = []
-        for modelname, species in zip(modelnames, species_ls):
-            conc = self._open_output_single_model(modelname, output, **settings)
-            if not isinstance(conc, GridDataArray):
-                raise RuntimeError(
-                    f"Type error. Expected GridDataArray but got {type(conc)}"
-                )
-            conc = conc.assign_coords(species=species)
-            concentrations.append(conc)
-        return concat(concentrations, dim="species")
+    def _concat_species(
+        self, output: str, species_ls: Optional[list[str]] = None, **settings
+    ) -> GridDataArray | GridDataset:
+        # groupby flow model, to somewhat enforce consistent transport model
+        # ordening:
+        # F1Ta1 F1Tb1 F2Ta2 F2Tb2 -> F1: [Ta1, Tb1], F2: [Ta2, Tb2]
```

Comment:
F1Ta1 F1Tb1 F2Ta2 F2Tb2 -> F1: [Ta1, Tb1], F2: [Ta2, Tb2]: this notation needs some explanation

Reply:
Added some explanation